### ISSUE TITLE
How to start multi-variate pre-training?

### ISSUE THREAD

**Issue by Leopold2333** [thuml/OpenLTM#13]
Seems that all pre-training Pytorch Dataset Classes select a single variable for fetching (including Dataset_ERA5_Pretrain)? Does Timer-XL allow for multivariate pretraining?

**Comment by WenWeiTHU** [comment]
Yes. I think it can be generally implemented under the supervised training for multivariate time series. However, the lack of a high-quality multivariate pretraining dataset can be the main obstacle.

**Comment by Leopold2333** [comment]
Get it. Has Timer-XL implemented the multivariate pre-training? Recently I came across some projects, take Moirai for example, it implemented a multivariate data loading process given varying datasets with different variate numbers. However, from Moirai's perspective, the amount of data in each dataset is determined by the number of the variables. This is different with Timer- models, where the amount of data is bound to the number of time points contained in a dataset. In this case, have any new multivariate data loading strategies been implemented?

**Comment by WenWeiTHU** [comment]
The current version is built on the Timer-XL structure (e.g. TimeAttention and RoPE), but this checkpoint is univariate pre-trained since the details of downstream tasks (e.g. the variate number) is agnostic. To make multivariate prediction, we currently recommend the channel-independence approach during inference. If you're interested in building domain-specific large models (e.g., the variate number can be fixed), we recommend Timer-XL for explicit multivariate pre-training. The example pipeline is conducted on the ERA5-MS on our codebase.

**Comment by Leopold2333** [comment]
Thank you for the reply. Does multivariate pretraining on ERA5-MS correspond to the following snippets in [data_loader.py](https://github.com/thuml/OpenLTM/blob/main/data_provider/data_loader.py)? I'm confused that all the dataloaders for pretraining seems returning univariate samples (except for the `MultivariateDatasetBenchmark`)?
https://github.com/thuml/OpenLTM/blob/610a5bfb2f6e69708a4e0a8de2c51688ad444f49/data_provider/data_loader.py#L385-L388
https://github.com/thuml/OpenLTM/blob/610a5bfb2f6e69708a4e0a8de2c51688ad444f49/data_provider/data_loader.py#L401-L404

**Comment by WenWeiTHU** [comment]
Yes, please use ```MultivariateDatasetBenchmark``` when the multivariate pre-training datasest is structurally unified. The training configuration is similiar to this [script](https://github.com/thuml/OpenLTM/blob/main/scripts/supervised/rolling_forecast/timer_xl_ecl.sh).
